llama_stack/distribution/library_client.py

```diff
@@ -210,49 +143,40 @@ def _remove_root_logger_handlers(self):
             root_logger.removeHandler(handler)
             print(f"Removed handler {handler.__class__.__name__} from root logger")
 
-    def _get_path(
-        self,
-        cast_to: Any,
-        options: Any,
-        *,
-        stream=False,
-        stream_cls=None,
-    ):
-        return options.url
-
     def request(self, *args, **kwargs):
-        path = self._get_path(*args, **kwargs)
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        async_response = loop.run_until_complete(
+            self.async_client.request(*args, **kwargs)
+        )
         if kwargs.get("stream"):
-            return stream_across_asyncio_run_boundary(
-                lambda: self.async_client.request(*args, **kwargs),
-                self.pool_executor,
-                path=path,
-                provider_data=self.provider_data,
-            )
-        else:
-
-            async def _traced_request():
-                if self.provider_data:
-                    set_request_provider_data(
-                        {"X-LlamaStack-Provider-Data": json.dumps(self.provider_data)}
-                    )
-                await start_trace(path, {"__location__": "library_client"})
+            # NOTE: We are using AsyncLlamaStackClient under the hood
+            # We need to convert the AsyncStream from async client into
+            # SyncStream return type for streaming
+            def sync_generator():
                 try:
```

Comment:
this `try` seems to be in the wrong place because the finally will close the loop but the generator is still alive and will hit an error in the next iteration then.

you also do need a try-catch around the non-stream case

Reply:
synced offline, updated to have more correct event loop handlings!